plotnine/scales/scale.py

```diff
@@ -284,7 +284,13 @@ def map_df(self, df):
 
         aesthetics = set(self.aesthetics) & set(df.columns)
         for ae in aesthetics:
-            df[ae] = self.map(df[ae])
+            mapped = self.map(df[ae])
+            if hasattr(mapped, 'shape') and len(mapped.shape) > 1:
+                # force pandas to treat it as tuples.
+                # Otherwise it uses the first entry of each tuple
+                # and throws away the rest
+                mapped = [tuple(x) for x in mapped]
+            df[ae] = mapped
```

Comment:
I think we can limit the change to the specific code path for manual linetypes. That is in `scale_linetype_manual.map`

```python
def map(...):
    result = super().map(...)
    # ...
    if len(result) and hasattr(result[0], '__hash__'):
        result = [tuple(x) for x in result]
    return result
```

This way there is no need to make changes to `guide_legend`.

Reply:
Sure. It's marginally more complicated since we still want to support mixed string/tuple linetype definitions, but it's done.